### DISCUSSION TITLE
Where can I get a GoCardless ID / Key Pair ?

### DISCUSSION THREAD

**Discussion by leokeba** [firefly-iii/firefly-iii#9027] [Firefly III]
Hello, I'm trying to setup GoCardless in the data importer, however it's asking for both an ID and a KEY and I can only create a "secret ID" in the user-secrets inferface. Am I missing something obvious ? There is no information about any kind of user ID or account ID in their API Docs.

**Comment by leokeba** [comment]
To be a little more specific, this is the URL where I can manage my secrets : [link](https://bankaccountdata.gocardless.com/user-secrets/)
And here's what I get :
<img width="1389" alt="Screenshot 2024-07-06 at 15 33 05" src="https://github.com/firefly-iii/firefly-iii/assets/31117813/03e78ede-57ae-4ac9-a83e-e86be99e3541">

**Reply by JC5** [reply]
Like it says when you press "Create new", the secret key will not be shown again after it has been created:

"This is a secret key for authentication. It will not be displayed later anywhere on the dashboard or website."

If you didn't store it at the moment you created it, you'll have to create a new one.

**Comment by PierfrancescoElia** [comment]
Hi @JC5, I am stuck in the same step as previous user.

From GoCardless Web UI, in that same page, I can obviously get the secret key, but the data importer's setup process also asks for "GoCardless Identifier", as per [this screenshot](https://docs.firefly-iii.org/tutorials/data-importer/gocardless/#import-from-gocardless_1), where can I get that? 

Thanks fro your kind help,
Pierfrancesco

**Reply by JC5** [reply]
Here is a screenshot.

<img width="1230" height="968" alt="Screenshot 2026-01-09 at 14 21 20" src="https://github.com/user-attachments/assets/1a5377d5-e079-4efb-8849-1af3e2457548" />

**Comment by PierfrancescoElia** [comment]
Dear @JC5,
thanks for your kind help, I managed to get it working, the base problem was that to login to GoCardless, I had to go to portal **https://bankaccountdata.gocardless.com** and login with Google (my old credentials were not working anymore, couldn't receive reset e-mail, idk). 

Logging from GoCardless.com home page takes the user to the screenshot @leokeba posted in previous message.

Now my GoCardless tokens work, but I have a new problem.

I have tried to connect my italian bank and a N26 bank account, both led to same problem. 

<img width="1119" height="764" alt="SCR-20260109-sdqk" src="https://github.com/user-attachments/assets/16cfd902-b274-4a4b-beab-7452a9bbef79" />

After allowing access with bank app, you can see in the screenshot that it looks like something is missing and can't join my accounts with Firefly via GoCardless. All my bank accounts are in Euro currency (that's configured and only currency activated in Firefly), for both of them I also set the correct IBAN code on Firefly app. My GoCardless/Nordigen API Key are set in .env file, and they seem to work (in fact I am able to complete bank account access).

What am I missing?

Thanks a lot for your awesome work.
Have a nice weekend!

**Reply by JC5** [reply]
What version of the data importer are you running?

**Reply by PierfrancescoElia** [reply]
Hi @JC5, latest version, **Firefly III Data Importer, v2.0.0**. Thanks :)

**Reply by JC5** [reply]
Try version 2.0.2. It has its own issues, but should list the accounts.

**Reply by PierfrancescoElia** [reply]
It worked! Thanks a lot!
